src/prefect/engine.py

```diff
@@ -1096,6 +1063,8 @@ async def begin_task_run(
                 anyio.create_task_group()
             )
 
+        await stack.enter_async_context(report_task_run_crashes(task_run, client))
```

Comment:
Investigate if this change can be split out

Reply:
No. This is needed because now we raise base exceptions instead of returning them.